core/trino-main/src/main/java/io/trino/sql/ExpressionUtils.java

```diff
@@ -255,6 +265,21 @@ public static Expression filterConjuncts(Metadata metadata, Expression expressio
         };
     }
 
+    public static Map<NodeRef<Expression>, Type> getExpressionTypes(PlannerContext plannerContext, Session session, Expression expression, TypeProvider types)
```

Comment:
There's already `TypeAnalyzer.getTypes()` for this purpose.

Reply:
Somehow `TypeAnalyzer` was not used in these places. Sounds like a miss.
Also, using `TypeAnalyzer` requires another injected dependency, as it's not part of `PlannerContext` (https://github.com/trinodb/trino/pull/10091#discussion_r772978698) and is hard to construct (https://github.com/trinodb/trino/pull/10091#discussion_r773017366).

Unifying the code, removing unnecessary existing copies of `getExpressionTypes`, is a good change.
If someone wants to replace this code with `TypeAnalyzer`, it will make the job easier.

Will add TODO for now.